### PR TITLE
docs: add nodejs quickstart guide

### PR DESCRIPTION
**Pull request by tinybeachthor**
Currently there is only a python quickstart guide, which is very nice.

This PR will add a similar guide for nodejs.

### CODE REVIEW

**Comment by DavHau** [comment]
Since dream2nix should work the same across ecosystems, I wonder if it would be best to just auto generate these quick start guides. We could have one `generic.md` containing the main text that just uses `{{#include ../subsystem-name.md}}` wherever it should say `python` or `nodejs` or whatever. Then we just need to create a new directory for each ecosystem containing the `./subsytem-name.md` and other files with the correct values. Then symlinking the `generic.md` file into each subsystem directory might be enough.

**Comment by tinybeachthor** [comment]
@DavHau having a unified guide for all ecosystems would be great eventually.
Right now, i would argue that there are enough idiosyncrasies and inconsistencies between different languages that it's beneficial to have a hand written guide for each one of them: with the ecosystem-specific vocabulary and conventions.

**Comment by DavHau** [comment]
This guide is really nice. Thanks.

> @DavHau having a unified guide for all ecosystems would be great eventually. Right now, i would argue that there are enough idiosyncrasies and inconsistencies between different languages that it's beneficial to have a hand written guide for each one of them: with the ecosystem-specific vocabulary and conventions.

I actually think that the guide of this PR already suitable to be used as generic guide. The `cowsay` package seems to exist in all ecosystems. Only words like `npm`, `nodejs`, `package.json`, `package-lock.json` would need to be replaced by the language specific variants.

File: docs/src/guides/getting-started-python.md [hunk 1]
```diff
@@ -30,16 +18,18 @@ In this example I will clone the python project [`httpie`](https://github.com/ht
 > nix flake init -t github:nix-community/dream2nix#simple
 wrote: /tmp/my_project/flake.nix
 ```
-Great, this created a new file `flake.nix` which is like a recipe that tells nix how to build our python project or how to assemble a development environment for it.  
+Great, this created a new file `flake.nix` which is like a recipe that tells nix how to build our python project or how to assemble a development environment for it.
 By modifying this file, we can tweak settings and change the way our package gets built by nix. But for now we just go with the defaults.
 
 ## Define the target platform
```
[the remark this answers is not in the document]
Comment (DavHau) [reply docs/src/guides/getting-started-python.md:24]: I mean the `Define the target platform` section.

**Pull request merged**